benchmarks/battleship/benchmark/src/main/java/com/android/designcompose/benchmark/battleship/benchmark/ClickPlayfield.kt

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.designcompose.benchmark.battleship.benchmark
+
+import androidx.benchmark.macro.ExperimentalMetricApi
+import androidx.benchmark.macro.StartupMode
+import androidx.benchmark.macro.TraceSectionMetric
+import androidx.benchmark.macro.TraceSectionMetric.Mode
+import androidx.benchmark.macro.junit4.MacrobenchmarkRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.uiautomator.By
+import androidx.test.uiautomator.Until
+import com.android.designcompose.DCTraces
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+/**
+ * This is a rough example of a benchmark, though one that doesn't focus on any behavior well.
+ * Essentially it benchmarks a large recomposition, by clicking the Opponent's Playfield button,
+ * which causes the full screen to re-draw.
+ */
+@OptIn(ExperimentalMetricApi::class)
+@RunWith(AndroidJUnit4::class)
+class ClickPlayfield {
+
+    @get:Rule val benchmarkRule = MacrobenchmarkRule()
+
+    @Test
+    fun changePlayers() =
+        benchmarkRule.measureRepeated(
+            packageName = PACKAGE_NAME,
+            metrics =
+                listOf(
+                    TraceSectionMetric(DCTraces.DESIGNDOCINTERNAL),
+                    TraceSectionMetric(DCTraces.DESIGNVIEW),
+                    TraceSectionMetric(DCTraces.DESIGNFRAME_DE_SUBSCRIBE, Mode.Sum),
+                    TraceSectionMetric(DCTraces.DESIGNFRAME_FINISHLAYOUT, Mode.Sum),
+                    TraceSectionMetric(DCTraces.DESIGNTEXT_DE, Mode.Sum),
+                    TraceSectionMetric(DCTraces.JNIADDNODES, Mode.Sum),
+                    TraceSectionMetric(DCTraces.DOCSERVER_DOC, Mode.Sum),
+                    TraceSectionMetric(DCTraces.DESIGNVIEW_INTERACTIONSCOPE, Mode.Sum),
+                ),
+            iterations = 5,
```

Comment:
5 is pretty low, good for development but I wouldn't draw any conclusions from it. To give some idea, I normally develop on ~3, make initial judgements on 10, and then before telling the team "I improved performance by x" I run it on 30.

Reply:
It can help to plot the results to see how noisy each iteration is, we have a WIP tool for it at https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:development/plot-benchmarks/;l=1?q=plot&sq=&ss=androidx

Reply:
Awesome, I needed something like that.

Reply:
> 5 is pretty low, good for development but I wouldn't draw any conclusions from it. To give some idea, I normally develop on ~3, make initial judgements on 10, and then before telling the team "I improved performance by x" I run it on 30.

Noted.